### PR TITLE
Adds badge with week # to events

### LINKED ISSUE #2396: Specifying competition week in team and competition page.
It'd be nice to see what week a competition is without checking my calendar.

### PR DESCRIPTION
**Pull request by jaredhasenklein**
Adds a badge with the Week number to an official event (or an event with a week number).
## Description
Closes #2396. If week number is not blank, adds the week number with special case for 2016.

## Motivation and Context
#2396 

## Screenshots (if appropriate):
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change API specifications or require data migrations)

### CODE REVIEW

**Comment by fangeugene** [comment]
I agree with @ZachOrr's suggestion in Slack that the week calculation code shouldn't exist in the template.

I suggest adding a `@property` in `event.py` [here](https://github.com/the-blue-alliance/the-blue-alliance/blob/master/models/event.py#L229) called `week_str` that calculates the week string.

Then, in the template, just check if `event.week_str is not None` and use the string.

**Comment by jaredhasenklein** [comment]
Sorry this has been such an ordeal, I have no idea what I'm doing but this is definitely helping me so hopefully I'll be able to do more advanced stuff in the future. Thanks all.

File: models/event.py [hunk 1]
```diff
@@ -226,7 +226,15 @@ def week(self):
             self._week = days / 7
 
         return self._week
-
+
```
Comment (dracco1993) [inline models/event.py]: `Blank line contains whitespace`
https://travis-ci.org/the-blue-alliance/the-blue-alliance/jobs/494044321#L3238

File: models/event.py [hunk 2]
```diff
@@ -226,7 +226,15 @@ def week(self):
...
-
+                          
+    @property
+    def week_str(self):
+        if self.week is None:
+            return None
+        if self.year == 2016:
+            return "Week {}".format(0.5 if self.week == 0 else self.week)
+        return "Week {}".format(self.week + 1)
+
```
Comment (dracco1993) [inline models/event.py]: `Blank line contains whitespace`
https://travis-ci.org/the-blue-alliance/the-blue-alliance/jobs/494044321#L3239

**Comment by jaredhasenklein** [comment]
Travis is complaining:
`jinja2.exceptions.TemplateAssertionError: no test named 'None'`

**Comment by jaredhasenklein** [comment]
@dracco1993 or @ZachOrr (or anyone else who sees this): I just opened this back up and started looking at the issue. Per [this](https://github.com/pallets/jinja/issues/520), the test should be `none` rather than `None` but looking in `event.py`, `None` is always capitalized. Is there another thing that could be causing Travis to spit out `jinja2.exceptions.TemplateAssertionError: no test named 'None'`?

File: templates_jinja2/event_details.html [hunk 3]
```diff
@@ -66,7 +66,7 @@ <h1 itemprop="summary">{{event.name}} {{event.year}}</h1>
           <span class="glyphicon glyphicon-circle-arrow-right"></span> Winners advance to the <a href="/event/{{ parent_event.key_name }}">{{ parent_event.name }}</a><br />
         {% endif %}
         {% if event.start_date %}
-          <span class="glyphicon glyphicon-calendar"></span> <time itemprop="startDate" datetime="{{event.start_date.isoformat()}}">{{ event.start_date|strftime("%B %d") }}{% if event.start_date.date() != event.end_date.date() %}</time> to <time itemprop="endDate" datetime="{{event.end_date.isoformat()}}">{{ event.end_date|strftime("%B %d") }}{% endif %}, {{ event.end_date|strftime("%Y") }}</time>
+          <span class="glyphicon glyphicon-calendar"></span> <time itemprop="startDate" datetime="{{event.start_date.isoformat()}}">{{ event.start_date|strftime("%B %d") }}{% if event.start_date.date() != event.end_date.date() %}</time> to <time itemprop="endDate" datetime="{{event.end_date.isoformat()}}">{{ event.end_date|strftime("%B %d") }}{% endif %}, {{ event.end_date|strftime("%Y") }}</time> {% if event.week_str is not None %}<span class="badge">{{event.week_str}}</span>{% endif %}
```
Comment (ZachOrr) [inline templates_jinja2/event_details.html]: Change this line to use the lowercase `none`, as you mentioned

File: models/event.py [hunk 4]
```diff
@@ -227,6 +227,14 @@ def week(self):
 
         return self._week
 
+    @property
+    def week_str(self):
```
Comment (ZachOrr) [inline models/event.py]: Can we write a quick test for this? Should be straightforward

File: templates_jinja2/event_details.html (hunk 3 above)
Comment (ZachOrr) [reply templates_jinja2/event_details.html]: Actually - can we not validate this by saying `{% if event.week_str %}`? That feels much better.

File: models/event.py (hunk 4 above)
Comment (jaredhasenklein) [reply models/event.py]: Opened #2401 to deal with this.

**Comment by fangeugene** [comment]
LGTM! Great work @jaredhasenklein and thanks everyone for input.

**Pull request merged**